### ISSUE TITLE
Error: TRANSMIT_COMPLETE_NO_ACK z-wave motion plus sensor

### ISSUE THREAD

**Issue by Sillec123** [athombv/com.fibaro#166]
Fibaro App v1.5.5
Homey: 1.5.0-rc.2

Hey i still get the following errors: 

[2017-08-05T06:57:12.383Z] ProcessSendData[221]: To node: 28 with data: 0x9880a41ed71f6310ce09 and txOptions: ACK,AUTO_ROUTE,EXPLORE
[2017-08-05T06:57:12.608Z] Command[221] start: sendData
[2017-08-05T06:57:22.054Z] ProcessSendData[221]: Marked node 28 as offline for the next 1800000ms
[2017-08-05T06:57:22.058Z] ProcessSendData[221]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-08-05T06:58:11.529Z] Command[221] end: sendData

[2017-08-05T07:02:21.356Z] Command[238] start: sendData
[2017-08-05T07:02:30.747Z] ProcessSendData[238]: To node: 28 with data: 0x9880b5f72b1fd9a75025 and txOptions: ACK,AUTO_ROUTE
[2017-08-05T07:02:30.755Z] Command[238] end: sendData
[2017-08-05T07:02:34.023Z] ProcessSendData[238]: Error: TRANSMIT_COMPLETE_NO_ACK

All parameters are default. I paired it again, changed the battery but the issue is still there.

**Comment by rnldnkp** [comment]
Same thing here

Homey 1.5.3
Fibaro 1.5.8

Motion sensor does not update status in Homey.
Have multiple and some even show nothing else but -

Have already rebooted homey and app separately, doesn't make a difference.

[2017-09-22T06:04:56.472Z] ProcessSendData[316]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-09-22T06:04:56.473Z] ProcessSendData[317]: To node: 2 with data: 0x988019e98c5e2ff79eb2 and txOptions: ACK,AUTO_ROUTE
[2017-09-22T06:04:56.476Z] Command[315] end: sendData
[2017-09-22T06:04:57.098Z] Node[2]: Received application command for COMMAND_CLASS_SECURITY, data: 0x40
[2017-09-22T06:04:57.099Z] Node[2]: sendData to COMMAND_CLASS_SECURITY, params 0x801a620dac0a38e21e
[2017-09-22T06:04:57.099Z] Command[317] start: sendData
[2017-09-22T06:04:57.155Z] ProcessSendData[317]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-09-22T06:04:57.158Z] ProcessSendData[318]: To node: 2 with data: 0x98801a620dac0a38e21e and txOptions: ACK,AUTO_ROUTE
[2017-09-22T06:04:57.160Z] Command[316] end: sendData
[2017-09-22T06:04:57.355Z] Node[2]: Received application command for COMMAND_CLASS_SECURITY, data: 0x40
[2017-09-22T06:04:57.356Z] Node[2]: sendData to COMMAND_CLASS_SECURITY, params 0x801be0f7b61a3cfb6b
[2017-09-22T06:04:57.356Z] Command[318] start: sendData
[2017-09-22T06:04:57.912Z] ProcessSendData[318]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-09-22T06:04:57.914Z] ProcessSendData[319]: To node: 2 with data: 0x98801be0f7b61a3cfb6b and txOptions: ACK,AUTO_ROUTE
[2017-09-22T06:04:57.917Z] Command[317] end: sendData
[2017-09-22T06:04:58.563Z] ProcessSendData[319]: Error: TRANSMIT_COMPLETE_NO_ACK
[2017-09-22T06:04:58.564Z] Command[318] end: sendData

**Comment by caseda** [comment]
Please try waking up your sensors manually

**Comment by rnldnkp** [comment]
I Will Try when i get home. Today I noticed homey starts to beep when a certain sensor is triggered. It sounds like and old motherboard with an error beep. Is that even possible!?

**Comment by rnldnkp** [comment]
Ok ignore my beep info, that's just a flow being triggered while it shouldn't. Homey App still has issues with home status, even after the App update... different story :)

For me the sensor with issues got reactived. Sadly Homey still ignores sometimes, other moments it works just fine. Do think this has nothing to do with the app. Did see many other reports starting 1.5.x, not limited to one app, but too multiple Z-wave apps.

**Comment by Veldkornet** [comment]
Not sure if this is related, bit if I look at the Z-Wave map, my multi-sensor is only ever connected to 1 device (and that's not Homey), never more, even if there are 6/7 devices around it. 
While all of the others connect to 2/3 devices at least.

**Comment by caseda** [comment]
@Veldkornet Since homey fw 1.5 the mesh map became pretty reliable, so it just means the sensor only can send data _through_ that device, or has never updated its internal neighboring Table
That can cause weirdness if that device it connects through doesn't report properly to homey. Does that device work properly?
But that can indeed be the same issue, bad connection = not being able to receive a acknowledge of that device

**Comment by RobinBol** [comment]
@Sillec123 did you try placing the sensor closer to Homey for some time and observe if that solves the 'no ack' issue? It is very likely a range issue.

**Comment by danone1973** [comment]
I have six Fibaro Motion Sensors Z-wave plus. If I include them then almost my whole z-wave network is flooded and most of my z-wave devices stop responding. The motion sensors themselves work randomly, one after another I get the below error (below Node22 is one of the Fibaro motion, this particular one located only two meters from Homey)
I hav tried everything I could think of. PTP, Reset even the z-wave network and add all again, soon as I start including this sensors, problems!
I did include them in HC2 and tried firmware update, but they were already with the latest firmware.
Simply I cannot use Homey since my z-wave is 60% down.
Once I exclude the Fibaro sensors, network is up and running again.

Anything I can do? I am more than happy to give someone my Homey login credentials to check what's going on!!

<img width="955" alt="screen shot 2017-10-21 at 9 09 01 pm" src="https://user-images.githubusercontent.com/30952373/31850853-07f1e846-b6a6-11e7-9c81-ee51ead97273.png">

**Comment by caseda** [comment]
just as for more information about this issue as we have another having the non homey issue.

the error "TRANSMIT_COMPLETE_NO_ACK" is homey telling that homey has send the information, but the sensor does _not report back_ that it received it.

This usually happens because the **distance is too big**, or because you _moved the sensor_ far from where you included it.
You can try to _wake up_ the sensor (by trippel pressing the button) but that **only** fixes the last thing i mentioned.
The distance is only _fixable_ by adding more mains powered z-wave devices to extend your range, and _testable_ with the way @RobinBol described, by putting the sensor closer to homey and waking it up once.

### Please try to do these 2 things (waking up on the new place and bringing it closer to homey) before adding on to this issue.

If you then wanted to say "but the home center 2 (or any other z-wave controller) had no trouble with that distance" then please also think about the 4 other (more) wireless rf signals homey can receive and send then most other z-wave only controllers.

**Comment by danone1973** [comment]
As per my post, I said that I tried everything. I should have mentioned that I had the sensors 2mm (millimetres) to Homey. Then I said that the sensor is 2m away. There are 6 sensors where the same issue seems to rotate in between then, now one works then the next minute doesn't and so on.
The biggest problem here is that when including those sensors most of my other z-wave devices simply stop responding. Then under z-wave, stuff for geeks, I get the TRANSMIT_COMPLETE_NO_ACK messages one after another and that's why I though I would have my input in this post. Sorry if I mistaken.
Maybe I should have opened another issue on github.

However, I was not going to say that HC2 had no trouble with the distance. Perhaps other members did say and you're upset now.  But now I am saying it. It certainly did not not have problems with distance, at least not when in between the HC2 and a battery powered device the distance was 2 meters!!!!!
I am also saying that Homey is great and I just want to make it work, and that I rely on nice and knowledgeable people like you for this to happen.
Thanks

P.S. I have a house with less than 100sqrm and have 31 powered z-wave devices spread all over. See my map. How many more shall I install??
<img width="730" alt="screen shot 2017-10-21 at 10 26 53 pm" src="https://user-images.githubusercontent.com/30952373/31851329-70392ac8-b6af-11e7-95df-705ab031edb2.png">

**Comment by caseda** [comment]
@danone1973 i only made my post because of more and more people getting this issue (more people using homey == more people with issues) and first try to blame homey, before doing anything themselves.
and enough people that _will_ say that their previous controller did work on that range (seen enough of those in the 1,5 year i'm working on z-wave now), so not because i'm angry, it's just to be ahead of people.

i still lack the "reading mind" option as one of my tools, so people not mentioning everything they have done (at least the 2 options minimal) it is very hard to determine what could be wrong.

but a small PS from me, to maybe fix/help what you knew about english (not to bash, just pre-cursor since i see it on all your light 😜), `light(s)` is written with the `t` at the end

**Comment by danone1973** [comment]
@caseda don't worry, thanks for your help here. I have been having this problem for two months now, I have been holding back and tried everything I could to find out the root of this problem, including reading on the forum and a z-wave book that I purchased from Amazon.
I think that it is either a network congestion problem or some routing problem which gets most of my z-wave devices not to respond anymore once the Fibaro Multisensors are added.
- no Fibaro multisensors in the network, everything works perfect and FAST.
- start to add the multisensors, and once I have more than two, troubles start.
- once I have all 6 multisensors included that I own, then more than half of my other z-wave devices either do not respond at all to commands (on/off) or they respond after several minutes!
Now my Homey looks like Saturn with all 6 Fibaro sensors around it like satellites. How much closer shall I get them?
I even installed an Aeotec Range Extender 6. It help greatly in getting a couple of door/window sensors which were far from any wired device, but that's all.

I would be more than happy to try any suggestion, as I said maybe someone can login into my Homey...

P.S. Indeed, I badly spel wrong and easy work. Yesterday I reset all my z-wave and then spent all Saturday adding all the devices back again. At last, in the evening I was so tired but I thougth I would rename them and mistaken perhaps the first word "light" from there copy/paste did the rest.
BTY, that inclusion was NOT easy, most of the times I had to try again and again as Homey kept looking up on different stages of the inclusion process. And yes, as I start include the Fibaro Multisensors, Homey start to tell me that "Wait for z-wave to be ready" and that can take for ever, I mean for ever unless PTP comes into action the z-wave does not get ready. Hence, I believe that there is some weird network congestion happening once those sensors come into it.

**Comment by danone1973** [comment]
I have done more testing as follow: (sorry for the long posts...)

1. Until Yesterday, I used to include the Fibaro multisensors then move them to their location.
2. Yesterday, I did fit a powerbank and I carried Homey around the house and include the Fibaro Multisensors at their final location.

On both the above cases, after including two sensors the network started to come up with errors and when all 6 of them the z-wave was in very bad shape with more than half of my devices not responding or responding after several seconds or even minutes.
Then on both of the above cases, I brought all the sensors around Homey, wake them up manually, but the errors stayed and my other z-wave devices still unresponsive.
Maybe worth mentioning that before I brought the sensors close to Homey, most of them were even not reporting movement or some of them stuck in tampering.
Also maybe worth mentioning that the z-wave devices that did not respond to any commands, it showed the change status if triggered manually, for example Fibaro Double Switch when I turned on/off the lights from the wall switch, the status changed accordingly in Homey desktop app.

**3. This morning I did the following: remove the Fibaro sensors (all my other z-wave devices worked perfect after that of course) then include them and left them around Homey (like Saturn...)
All seem to work for the first time super! All six sensors included, no errors could be seen, reporting all breach, tampering, temperature, lux. Also all my other z-wave devices working perfect and responding fast.**

After the above point 3 I left home and returned after 4 hours. Tested and cried... Again, more than half of my z-wave network was unresponsive. Then interesting fact, I looked at the Fibaro multisensors status since I had just to wave my hand above all 6, and learned that only three of them were reporting motion!! The rest, no. I tried this for several minutes, same result. Then I start to shake one after another and check their status while doing that. Surprised, the three sensors that did not respond motion after shacking, still no motion reported but the tampering alarm came on.
In that moment I took the below screenshot.
Eventually, after few more minutes out of the blue, the three multisensors started to report again motion. But the other z-wave devices don't respond....
Remember, I did not take away from Homey the multisensors all this time!!

It looks like a network congestion caused by some bad routing. I guess that first when I paired the multisensors and kept them close to Homey, they were communicating direct with Homey hence, no problems with my other devices.
Then after I left them alone for several hours, the z-wave mesh network happened and now the multisensors were communicating with Homey via many other z-wave devices and blocked them eventually.
**I noticed that every time the same z-wave devices stop responding, sometime more of them, other times less. Those devices are like Fibaro Double Switch 2, Fibaro Dimmer 2, Aeotec Switch 6 and 5. Never the Fibaro RGBW or Fibaro Double Relay 2 or Fibaro Motor Control 2 had a problem... I think that all this without a response problem are actually z-wave protocol and not z-wave plus.**

I want to try one more thing, that's to get some of the above z-wave plus devices that stop responding to a friend with an HC2 and try a firmware update. (the Fibaro multisensors have definitely the latest firmware)
Unfortunately next week I am on a business trip and have no time for this operation.

Below you see a screenshot taken on point 3, the Node 27, 17 and 22 were the three Fibaro Multisensors that were not reporting anymore the motion.

<img width="939" alt="screen shot 2017-10-22 at 2 33 16 pm" src="https://user-images.githubusercontent.com/30952373/31858161-62b5afec-b73c-11e7-9ef0-5ea2a5d04f88.png">

**Comment by ArchiSpec** [comment]
@danone1973 similar situation here regarding Fibaro motion sensors continually reporting tamper and only sometimes working. Thus thought the the whole issue with Fibaro units might be connected to the motion sensors. However, have removed them from Homey and still problem with fibaro units persists. So they are not causing the problem as far as I can ascertain. I have kept them out of the network for a couple of days, and still having same problems. I have been replacing all fibaro units with qubino to "solve" the problem. No issues with them, so its definitely specific to fibaro.

**Comment by addwindtan** [comment]
@danone1973

Great work with the troubleshooting. I’m facing the same problems with the Fibaro motion sensor too.

@jeroenvollenbrock

Hope to see a fix for this soon. The Fibaro motion sensors are common here in Singapore and a bug like this will have great impact to the adopters here.

**Comment by Lowlander61** [comment]
I had to remove all my fibaro motion sensors to get my z-wave running again. I too did some testing, with only 1 motion sensor network was just oke, some delay occurred, but after adding 1 or more motion sensor the whole z-wave network was down. For the time being I replaced a few of the sensor with the Neo Coolcam motion sensors and they are working perfect. At the moment, without the fibaro, a fast z-wave network!

**Comment by addwindtan** [comment]
This issue is tagged as "Waiting for user input". May I know what user input is required? I could try to provide inputs to get this issue moving.

**Comment by caseda** [comment]
it is the question he asked just before it :wink:
> @Sillec123 did you try placing the sensor closer to Homey for some time and observe if that solves the 'no ack' issue? It is very likely a range issue.

**Comment by addwindtan** [comment]
@RobinBol 

After I included the FIBARO motion sensor and place it right beside Homey, I do not notice any 'no ack' issues.

When I move the motion sensor to another location out of range of Homey where it depends on other FIBARO devices to repeat the signal, I start to observe 'No ack' issues.

When I move the motion sensor back beside Homey, the 'No ack' issues persisted.

It stops after I have removed it from my Z-Wave network.

I hope this helps you in resolving this critical issue.

**Comment by caseda** [comment]
@addwindtan did you wake it up when on the new spot, but also when putting it back again close to homey?

**Comment by addwindtan** [comment]
@caseda Yes, I did wake it up when I put it close to Homey.

**Comment by Veldkornet** [comment]
With regard to the range of the motion sensor which may be related to this problem, originally when I added it, it connected to multiple devices ( around 5 according to the Z-Wave Map); including Homey which is at max 3,5m away.

1 by one the connections dropped off though, and now only 1 remains. It only connects to the Fibaro Wall Plug 2 (which is about 1m away) and nothing else.

**Comment by addwindtan** [comment]
It sounds like the FIBARO motion sensor is a virus to our Homey.

**Comment by ArchiSpec** [comment]
@addwindtan  indeed!!

Can I suggest (at risk of upsetting @caseda (again!)) that this is exactly the same issue as https://github.com/athombv/homey/issues/1899 which was opened as a master case to pull all these issues together.

Just to add though, that I have continued testing as has @danone1973, @bm480 and others have done. A few points to hopefully assist @caseda, @RobinBol & @WeeJeWel in their hard work to overcome this known issue:-

1. I believe it is the motion sensors causing the problem (in my case FGSM-001 z-wave plus)
2. BUT, the motion sensors keep reporting correctly (movement, temp & humidity - tamper seems to be more of a problem) but appear to kick out all the other Fib units
3. In other words, sensors seem to cause the problem with all Fib dimmer module and SOME switches, but keep reporting correctly themselves
4. My Client install where we ripped out 20+ Fib units and replaced with Qubino continues to operate with no problems!!
5. I have been adding all the removed Fib units into my test rig at home. No problem with dimmer & switch modules. THEN I add 1 sensor, no problem (including moving it out of range as well, meshed off mains unit)
6. THEN I add a SECOND sensor and keep it RIGHT BY HOMEY!!! - and just like @Lowlander61 above, all Fib units stop working again. EVERYTHING else is fine, and sensors still continue reporting correctly.
7. REMOVE that last sensor, and hey presto, everything works again.

I make no other observations, just to give my experience for those looking into it. I'll repost this on the main issue https://github.com/athombv/homey/issues/1899 just for ease.
